src/databricks/labs/ucx/source_code/graph.py

```diff
@@ -212,14 +252,19 @@ class DependencyGraphContext:
 
 class Dependency(abc.ABC):
 
-    def __init__(self, loader: DependencyLoader, path: Path):
+    def __init__(self, loader: DependencyLoader, path: Path, inherits_context=True):
```

Comment:
```suggestion
    def __init__(self, loader: DependencyLoader, path: Path, inherits_context: bool = True):
```

Reply:
not sure that's good practice ? Can you provide a reference ?